### PR TITLE
[Enhancement] Add queue_subscription support for nats jetstream streaming source

### PR DESCRIPTION
**Pull request by atjhoendz**
# Description

https://github.com/mage-ai/mage-ai/issues/4122 implementation, a configurable nats jetstream subscription method, that support for PULL and PUSH consumer type with queue group options.

New Configuration:
- consumer_type: define what the type of consumer you want, available types: PULL and PUSH, the default is PULL.
- use_queue_group: this is the configuration for a PUSH consumer. If set to True, it allows the consumer to use a queue group.

# How Has This Been Tested?

- tested locally by creating NATS streaming pipeline with new configuration.

# Checklist
- [x] The PR is tagged with proper labels (bug, enhancement, feature, documentation)
- [x] I have performed a self-review of my own code
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] If new documentation has been added, relative paths have been added to the appropriate section of `docs/mint.json`

cc:

@mfreeman451 @wangxiaoyou1993

### CODE REVIEW

**Review by mfreeman451 — APPROVED** [review]

**Comment by aslamhadi** [comment]
hi, any updates on this? I also need this feature on my side. thanks a lot before

File: mage_ai/streaming/sources/nats_js.py [hunk 1]
```diff
@@ -21,6 +21,12 @@ class SSLConfig:
     check_hostname: bool = False
 
 
+@dataclass
+class ConsumerType:
```
Comment (wangxiaoyou1993) [inline mage_ai/streaming/sources/nats_js.py:25]: The type of this class should be enum.Enum. It doesn't need the dataclass decorator.
```
class ConsumerType(str, enum.Enum):
```

File: mage_ai/streaming/sources/nats_js.py [hunk 2]
```diff
@@ -165,24 +196,36 @@ def batch_read(self, handler: Callable):
             self.stop_loop()
 
     def read(self, handler: Callable):
-        self.init_client()
```
Comment (wangxiaoyou1993) [inline mage_ai/streaming/sources/nats_js.py:168]: why do you not call `init_client`?

File: mage_ai/streaming/sources/nats_js.py (hunk 1 above)
Comment (atjhoendz) [reply mage_ai/streaming/sources/nats_js.py]: thx, updated 🙌🏻

File: mage_ai/streaming/sources/nats_js.py (hunk 2 above)
Comment (atjhoendz) [reply mage_ai/streaming/sources/nats_js.py:168]: because it's already called by the init method of BaseSource class

**Review by mfreeman451 — APPROVED** [review]

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**